### ISSUE TITLE
AdGuard breaks Google sync

### ISSUE THREAD

**Issue by djdelarosa25** [AdguardTeam/AdguardForAndroid#2925]
### Steps to reproduce

1. Install AdGuard, don't configure anything. Just enable it and accept the prompt to enable VPN.
2. Go to Settings>Accounts>tap on your Google account>Account sync>tap on the three-dot icon on the top right>Sync now

### Expected behavior
All parameters should sync instantly.

### Actual behavior
Only the Google Fit data and People details parameters actually sync. The rest get stuck, as evidenced by the two black arrows not spinning and the time and date of last sync not updating.

<details><summary>Screenshot:</summary>

![Screenshot_20190715-103901_Settings](https://user-images.githubusercontent.com/34760155/61193305-2d5c6280-a6ed-11e9-9d41-14f1c8e1724f.png)

</details>

### Your environment

* AdGuard version: v3.1.0 (1.3.347cl)
* AdGuard filtering mode: Local VPN
* Device model: Xiaomi Poco F1
* Operating system and version: LineageOS 16 (Android 9)
* Root access: Yes
* Any other apps that can start VPN / Antiviruses / Cleaners etc. installed?: None

**Comment by zzebrum** [comment]
@TheHasagi please check

**Comment by TheHasagi** [comment]
@djdelarosa25 

Strange, unable to reproduce. Please, check with the latest [nightly build](agrd.io/android_nightly).

**Comment by djdelarosa25** [comment]
I did, no dice. Whitelisting the apps in the firewall worked, though.

**Comment by Eugene-Savenko** [comment]
Another report here: 1555536

> On our Samsung Galaxy S8's (latest updates), when AdGuard is running, Google contacts won't sync. When I disable AdGuard, sync works. Google Backup also does not work with AdGuard running.

**Comment by ameshkov** [comment]
@Eugene-Savenko to troubleshoot this issue, we need to get additional logs.

Here's what we need you to do:

1. Collect the debug log as it's [explained here](https://kb.adguard.com/en/android/solving-problems/log#debug).
2. Remember the exact time when the issue was reproduced, we will need it to find the corresponding records in the log file.
3. Additionally, capture the system bug report after reproducing the issue. [Here's the instruction](https://developer.android.com/studio/debug/bug-report).
4. Send both files to `devteam@adguard.com`. Mention the Github issue number and the exact time when the issue was reproduced.

**Comment by djdelarosa25** [comment]
For a while, I got it working after clean flashing LineageOS 16 again. However, just yesterday, I switched to Pixel Experience and the problem has re-emerged.

**Comment by ameshkov** [comment]
@djdelarosa25 it'd help if you can collect the logs as per https://github.com/AdguardTeam/AdguardForAndroid/issues/2925#issuecomment-518012657

**Comment by djdelarosa25** [comment]
Issue got fixed for me by updating to the latest beta (3.2.121).

**Comment by TheHasagi** [comment]
@djdelarosa25 

You should check with the release version (3.2.135).

**Comment by TheHasagi** [comment]
As per user did not reply, no evidence provided the issue is closed. Will be reopened if needed.

**Comment by TPS** [comment]
#5617?

**Comment by maxikuzmin** [comment]
@TPS In order to troubleshoot this issue, we need to get your application logs.

Here's what we need you to do:

1. Enable debug logging:
*Settings -> General -> Advanced -> Logging level -> Debug*.
2. Reproduce the problem, then remember the exact time when it happened.
3. Collect logs:
*Settings -> General -> Advanced -> Export logs and system info*.
4. Send this file to [qa@adguard.com](mailto:qa@adguard.com):
    - include `[android]` keyword and `2925` in the subject of your email
    - specify the exact time when the issue occurred


Also, please do not forget to check Recent activity and view blocked requests. You can also send screenshots to your email

**Comment by TPS** [comment]
@maxikuzmin Did y'all get you needed from me?

**Comment by maxikuzmin** [comment]
@TPS no, I don't see any logs from you. Please try sending them again as instructed

**Comment by TPS** [comment]
@maxikuzmin Resent 6 hours ago.

**Comment by maxikuzmin** [comment]
@TPS thank you! We will need some time to check your logs

**Comment by maxikuzmin** [comment]
@TPS we are working on it, please wait

**Comment by maxikuzmin** [comment]
@TPS we need a little more time, please wait

**Comment by rsadikov01** [comment]
@TPS Hello! We've tried to reproduce this issue on Samsung, Huawei and Xiaomi phones but it wasn't worked out. So we need your help.
As I can see from your logs, you have an AdGuard VPN. However, I couldn't figure out whether the integration mode was enabled in AdGuard for Android along with AdGuard VPN.
You can check it in your VPN application:
**Settings -> General -> Advanced -> Operating Mode** (VPN, SOCKS5 or Integrated mode)

If Google Sync isn't working with both AdGuard VPN in integration mode and AdGuard for Android enabled, please follow the steps described above by maxikuzmin for both apps, like you did before. Then send logs from both apps to the email you've sent it before and specify the exact time when the issue occurred
Also would be useful if you'll provide me this kind of information:
1. Do you use Wi-Fi or mobile network?
- if Wi-Fi — which provider
- if mobile — operator + country
2. Which location is selected in AdGuard VPN (country/city)?
3. Is battery saver mode enabled on your mobile phone when this issue occurs?
